File: client/decentralized_comm_client/lib/models/message.dart
class Message {
  final int id;
  final int chatId;
  final int senderId;
  final String content;
  final String createdAt;

  Message({
    required this.id,
    required this.chatId,
    required this.content,
    required this.senderId,
    required this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) => Message(
    id: json['id'],
    chatId: json['chat_id'],
    senderId: json['sender_id'],
    content: json['content'],
    createdAt: json['created_at'],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "chat_id": chatId,
    "sender_id": senderId,
    "content": content,
    "created_at": createdAt,
  };
}

File: server/decentralized_comm_server/app/models/message.py
from datetime import datetime
from sqlalchemy.sql import func
from app.extensions import db


class Message(db.Model):
    __tablename__ = "messages"

    id = db.Column(db.Integer, primary_key=True)
    chat_id = db.Column(db.Integer, db.ForeignKey("chats.id"), nullable=False)
    sender_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable = False)
    content = db.Column(db.Text, nullable = False)
    created_at = db.Column(db.DateTime(timezone = True), server_default = func.now())

    def __repr__(self):
        return f"<Message {self.id}  chat={self.chat_id} sender={self.sender_id}>"